code/datums/components/crafting/recipes.dm

```diff
@@ -41,6 +41,18 @@
 		tool_behaviors = string_list(tool_behaviors)
 	if(tool_paths)
 		tool_paths = string_list(tool_paths)
+	for(var/item in blacklist)
+		var/actual_item = new item(null)
+		if(!isitem(actual_item) || item == result)
+			continue
+		if(!(item in GLOB.crafting_recipe_items_blacklist) && !(item in GLOB.crafting_recipe_items))
+			GLOB.crafting_recipe_items_blacklist += item
+	for(var/item in reqs)
+		var/actual_item = new item(null)
+		if(!isitem(actual_item))
+			continue
+		if(!(item in GLOB.crafting_recipe_items))
+			GLOB.crafting_recipe_items += item
```

Comment:
bro what

Reply:
The item might already be in the GLOB list, since it's a for loop and it might have been added while looping through a previous recipe